assets/src/edit-story/components/canvas/canvasLayout.js

```diff
@@ -1,62 +1,223 @@
+	height: 100%;
+`;
+
+const DisplayLayer = styled.div`
+	${ Layer }
+	${ Layout }
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(0, 255, 0, 0.2);
+		transform: translateZ(200px);
+	` ) }
+`;
+
+const DisplayCanvas = styled.div`
+	${ Canvas }
+	grid-area: page;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(255, 255, 255, 0.2);
+	` ) }
+`;
+
+const NavLayer = styled.div`
+	${ Layer }
+	${ Layout }
+	pointer-events: none;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(0, 0, 255, 0.2);
+		transform: translateZ(400px);
+	` ) }
 `;
 
-const Area = styled.div`
+const NavArea = styled.div`
+	background-color: ${ ( { theme } ) => theme.colors.bg.v1 };
 	grid-area: ${ ( { area } ) => area };
 	height: 100%;
 	width: 100%;
+	pointer-events: initial;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(0, 0, 255, 0.4);
+		transform: translateZ(400px);
+	` ) }
+`;
+
+const FramesAndSelectionLayer = styled.div`
+	${ Layer }
+	${ Layout }
+	pointer-events: none;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(0, 255, 255, 0.2);
+		transform: translateZ(600px);
+	` ) }
+`;
+
+const Overlay = css`
+	position: absolute;
+	top: 0;
+	left: 0;
+	width: 0;
+	height: 0;
+
+	${ DEBUG_ANGLE && ( `
+		width: 100px;
+		height: 100px;
+	` ) }
+`;
+
+const FramesOverlay = styled.div`
+	${ Overlay }
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(255, 0, 255, 0.2);
+	` ) }
+`;
+
+const SelectionOverlay = styled.div`
+	${ Overlay }
+
+	${ DEBUG_ANGLE && ( `
+		top: 20px;
+		left: 20px;
+		background-color: rgba(255, 255, 0, 0.2);
+	` ) }
+`;
+
+// QQQQ: only show graypane for few element types.
+const EditLayer = styled.div`
+	${ Layer }
+	${ Layout }
+	pointer-events: none;
+	background-color: rgba(0, 0, 0, 0.5);
+	z-index: 1;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(255, 140, 50, 0.2);
+		transform: translateZ(800px);
+	` ) }
 `;
 
 function CanvasLayout() {
+	const {
+		state: { currentPage },
+	} = useStory();
+	const {
+		state: { editingElement: editingElementId },
+	} = useCanvas();
+	// eslint-disable-next-line @wordpress/no-unused-vars-before-return
+	const editingElement =
+    editingElementId &&
+    currentPage &&
+    currentPage.elements.find( ( element ) => element.id === editingElementId );
+
 	return (
-		<SelectionCanvas>
-			<Background>
-				<Area area="page">
-					<Page />
-				</Area>
-				<Area area="menu">
-					<PageMenu />
-				</Area>
-				<Area area="prev">
-					<PageNav isNext={ false } />
-				</Area>
-				<Area area="next">
-					<PageNav />
-				</Area>
-				<Area area="carrousel">
-					<Carrousel />
-				</Area>
-			</Background>
-		</SelectionCanvas>
+		<WorkspaceBackground>
+			<SelectionCanvas>
+				<DisplayLayer>
+					<DisplayCanvas>
+						<PageDisplay />
+					</DisplayCanvas>
+				</DisplayLayer>
+				<NavLayer onMouseDown={ ( evt ) => evt.stopPropagation() }>
+					<NavArea area="menu">
+						<PageMenu />
+					</NavArea>
+					{ /* QQQ: buttons are too big. especially given the pointerEvents
```

Comment:
#4106 is gonna improve the layout there.

In the meantime, this could be a quick fix:


```patch
 	display: grid;
 	grid:
-    ".    .      . .          . .        ." 1fr
-    ".    prev   . page       . next     ." ${ PAGE_HEIGHT }px
-    ".    .      . menu       . .        ." 48px
-    ".    .      . .          . .        ." 1fr
-    ".    carrousel      carrousel carrousel  carrousel carrousel        ." 60px
-    / 1fr ${ LEFT_NAV_WIDTH }px ${ PAGE_NAV_BUTTON_WIDTH }px ${ PAGE_WIDTH }px ${ PAGE_NAV_BUTTON_WIDTH }px ${ LEFT_NAV_WIDTH }px 1fr;
+    ".         .         .         .         .         " 1fr
+    ".         prev      page      next      .         " ${ PAGE_HEIGHT }px
+    ".         .         menu      .         .         " 48px
+    ".         .         .         .         .         " 1fr
+    "carousel  carousel  carousel  carousel  carousel  " 60px
+    / 1fr ${ LEFT_NAV_WIDTH }px ${ PAGE_WIDTH }px ${ LEFT_NAV_WIDTH }px 1fr;
```

Reply:
I'll just sync.